### PR TITLE
Introduce jgiven-html-app as dependency

### PR DESCRIPTION
**Pull request by l-1squared**
Schau mal, ich habe unseren Riesenpfusch in einen etwas offensichtlicheren Pfusch verwandelt

### CODE REVIEW

File: src/reportAppAzureDirective.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+                });
+            } else {
+                elem.on('click', function (e) {
+                    e.preventDefault();
+                    window.open('https://jgiven.org', '_blank');
+                });
+            }
+        }
+    }
+})
```
Comment (fudler) [inline src/reportAppAzureDirective.js:29]: The idea is nice :) 
Although I guess there is quite some more code inside our app.bundle.js. 
As far as I can see we need all the stuff from line 98524 to line 98611.

I am not sure whether all lines in between are needed or whether there is any duplicate, although I will check this!

**Review by fudler — CHANGES_REQUESTED** [review]

File: src/reportAppAzureDirective.js (hunk 1 above)
Comment (l-1squared) [reply src/reportAppAzureDirective.js:29]: Damn, I just saw that this code is not on top-level, but inside a function! That makes everything worse.

Comment (fudler) [reply src/reportAppAzureDirective.js:29]: We know which function is manipulated, so maybe it is possible to insert the necessary code via some fancy usage of sed. However, a 'nice' solution is something different indeed.

Comment (l-1squared) [reply src/reportAppAzureDirective.js:29]: I might havefound a solution. But it is ten times more hacky :(

Comment (fudler) [reply src/reportAppAzureDirective.js:44]: Before merging I am going to test the extension in DevOps :)

Comment (fudler) [reply src/reportAppAzureDirective.js:44]: After removing a few unnecessary dependencies the plugin builds and a test pipeline succeeded.

However, Azure DevOps won't display the results anymore (stuck in an endless loading procedure). 
I will investigate whether this is due to some lazyness on AzureDevops-Side or whether we destroyed some important functionality with this PR.

Therefore, for now we should not merge.

File: scripts/prebuild.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+SCRIPT_LOCATION=$(dirname -- "$(readlink -f -- "${BASH_SOURCE[0]}")")
+export TOPLEVEL="${SCRIPT_LOCATION}/.."
```
Comment (fudler) [inline scripts/prebuild.sh:3]: By the way, what does this do? The "location/.."?

File: src/reportAppAzureDirective.js (hunk 1 above)
Comment (l-1squared) [reply src/reportAppAzureDirective.js:44]: you mean node_modules/jgiven-html-app/dist? yes, that should be filled by the yarn build step! if that does not happen the app won't be copied to the toplevel.

Comment (fudler) [reply src/reportAppAzureDirective.js:44]: Exactly, this folder seems to stay empty. The build seems to be triggered although this folder stays empty for some reason.

Comment (fudler) [reply src/reportAppAzureDirective.js:44]: Funnily enough, using an older version like 0.16.2 includes the needed files.

Comment (fudler) [reply src/reportAppAzureDirective.js:43]: The yarn build command will generate a new app.bundle.js which is actually a lot smaller than the one delivered via npm install.
I suspect there are few missing commands in this new app.bundle.js which lead to the malfunctioning Azure plugin. Therefore we should maybe think about some sort of PostBuild script to manipulate the app.bundle.js directly, although this also seems to be rather annoying.

Comment (l-1squared) [reply src/reportAppAzureDirective.js:43]: weird...

**Comment by fudler** [comment]
Closed due to inherent incompatibilities. 
The HTML App does not build as intended and all attempts to resolve this only lead to more difficulties. We will try a new attempt once the HTML App was reforged.

**Pull request closed without merging**